docs/reference/modules/discovery/publishing.asciidoc

```diff
@@ -0,0 +1,39 @@
+[[cluster-state-publishing]]
+=== Cluster state publishing
+
+The master node is the only node in a cluster that can make changes to the
+cluster state. The master node processes one batch of cluster state updates at
+a time, computing the required changes and publishing the updated cluster state
+to all the other nodes in the cluster. Each publication starts with the master
+broadcasting the updated cluster state to all nodes in the cluster, to which
+each node responds with an acknowledgement but does not yet apply the
+newly-received state. Once the master has collected acknowledgements from
+enough master-eligible nodes the new cluster state is said to be _committed_,
+and the master broadcasts another message instructing nodes to apply the
+now-committed state. Each node receives this message, applies the updated
+state, and then sends a second acknowledgement back to the master.
+
+The master allows a limited amount of time for each cluster state update to be
+completely published to all nodes, defined by `cluster.publish.timeout`, which
+defaults to `30s`, measured from the time the publication started. If this time
+is reached before the new cluster state is committed then the cluster state
+change is rejected, the master considers itself to have failed, stands down,
+and starts trying to elect a new master.
+
+However, if the new cluster state is committed before `cluster.publish.timeout`
+has elapsed, but before all acknowledgements have been received, then the
```

Comment:
```suggestion
has elapsed, the
```

Reply:
I see the point of this change but it's not quite what I was trying to say. The master waits for acks, up to the timeout, before proceeding with the next update. I tried different wording in 442a7a76934.